### PR TITLE
[Enhancement] support compile with jemalloc

### PR DESCRIPTION
**Pull request by eldenmoon**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by dataroaring — DISMISSED** [review]

**Comment by morningman** [comment]
hi @eldenmoon , please rebase to pass the check.
And also, we need to modify the compile doc to explain how to use it.

File: be/CMakeLists.txt [hunk 1]
```diff
@@ -487,6 +487,8 @@ elseif ("${CMAKE_BUILD_TYPE}" STREQUAL "RELEASE")
     SET(CMAKE_CXX_FLAGS ${CXX_FLAGS_RELEASE})
 elseif ("${CMAKE_BUILD_TYPE}" STREQUAL "ASAN")
     SET(CMAKE_CXX_FLAGS "${CXX_FLAGS_ASAN}")
+elseif ("${CMAKE_BUILD_TYPE}" STREQUAL "JEMALLOC")
```
Comment (yiguolei) [inline be/CMakeLists.txt:490]: Use a separate flags to indicate using jemalloc or not.

Comment (eldenmoon) [reply be/CMakeLists.txt:490]: OK

File: be/src/http/action/pprof_actions.cpp [hunk 2]
```diff
@@ -60,7 +60,8 @@ class HeapAction : public HttpHandler {
 
 void HeapAction::handle(HttpRequest* req) {
     std::lock_guard<std::mutex> lock(kPprofActionMutex);
-#if defined(ADDRESS_SANITIZER) || defined(LEAK_SANITIZER) || defined(THREAD_SANITIZER)
+#if defined(ADDRESS_SANITIZER) || defined(LEAK_SANITIZER) || defined(THREAD_SANITIZER) || \
+        defined(USE_JEMALLOC)
     (void)kPprofDefaultSampleSecs; // Avoid unused variable warning.
 
     std::string str = "Heap profiling is not available with address sanitizer builds.";
```
Comment (yiguolei) [inline be/src/http/action/pprof_actions.cpp:67]: Should modify the error message, add jemalloc to hint.

File: be/src/http/action/pprof_actions.cpp [hunk 3]
```diff
@@ -110,7 +111,8 @@ class GrowthAction : public HttpHandler {
 };
 
 void GrowthAction::handle(HttpRequest* req) {
-#if defined(ADDRESS_SANITIZER) || defined(LEAK_SANITIZER) || defined(THREAD_SANITIZER)
+#if defined(ADDRESS_SANITIZER) || defined(LEAK_SANITIZER) || defined(THREAD_SANITIZER) || \
+        defined(USE_JEMALLOC)
     std::string str = "Growth profiling is not available with address sanitizer builds.";
```
Comment (yiguolei) [inline be/src/http/action/pprof_actions.cpp:116]: Here, add jemalloc to hint.

File: be/src/http/action/pprof_actions.cpp [hunk 4]
```diff
@@ -132,7 +134,8 @@ class ProfileAction : public HttpHandler {
 };
 
 void ProfileAction::handle(HttpRequest* req) {
-#if defined(ADDRESS_SANITIZER) || defined(LEAK_SANITIZER) || defined(THREAD_SANITIZER)
+#if defined(ADDRESS_SANITIZER) || defined(LEAK_SANITIZER) || defined(THREAD_SANITIZER) || \
+        defined(USE_JEMALLOC)
     std::string str = "CPU profiling is not available with address sanitizer builds.";
```
Comment (yiguolei) [inline be/src/http/action/pprof_actions.cpp:139]: Here, add jemalloc to hint.

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**